test/cartesian/CartesianGrid.spec.tsx

```diff
@@ -30,6 +30,179 @@ describe('<CartesianGrid />', () => {
+      expect.soft(background).toHaveAttribute('height', '200');
+    });
+
+    it('should put x, y, width and height as coordinates to all lines', () => {
+      const { container } = render(
+        <Surface width={500} height={500}>
+          <CartesianGrid
+            x={1}
+            y={2}
+            width={100}
+            height={200}
+            fill="green"
+            verticalPoints={verticalPoints}
+            horizontalPoints={horizontalPoints}
+          />
+        </Surface>,
+      );
+      const horizontalLines = container.querySelectorAll('.recharts-cartesian-grid-horizontal line');
+      expect(horizontalLines).toHaveLength(5);
+      horizontalLines.forEach(line => {
+        expect.soft(line).toHaveAttribute('x', '1');
+        expect.soft(line).toHaveAttribute('y', '2');
+        expect.soft(line).toHaveAttribute('x1', '1');
+        expect.soft(line).toHaveAttribute('x2', '101');
+      });
+      const verticalLines = container.querySelectorAll('.recharts-cartesian-grid-vertical line');
+      expect(verticalLines).toHaveLength(4);
+      verticalLines.forEach(line => {
+        expect.soft(line).toHaveAttribute('x', '1');
+        expect.soft(line).toHaveAttribute('y', '2');
+        expect.soft(line).toHaveAttribute('y1', '2');
+        expect.soft(line).toHaveAttribute('y2', '202');
+      });
+    });
+
+    it('should put x, y, width and height as coordinates to all stripes', () => {
+      const { container } = render(
+        <Surface width={500} height={500}>
+          <CartesianGrid
+            x={1}
+            y={2}
+            width={100}
+            height={200}
+            fill="green"
+            verticalPoints={verticalPoints}
+            horizontalPoints={horizontalPoints}
+            horizontalFill={['red', 'black']}
+            verticalFill={['green', 'blue']}
+          />
+        </Surface>,
+      );
+      const horizontalStripes = container.querySelectorAll('.recharts-cartesian-gridstripes-horizontal rect');
+      expect(horizontalStripes).toHaveLength(5);
+      horizontalStripes.forEach(stripe => {
+        expect.soft(stripe).toHaveAttribute('x', '1');
+        expect.soft(stripe).toHaveAttribute('width', '100');
+      });
+      const verticalStripes = container.querySelectorAll('.recharts-cartesian-gridstripes-vertical rect');
+      expect(verticalStripes).toHaveLength(4);
+      verticalStripes.forEach(stripe => {
+        expect.soft(stripe).toHaveAttribute('y', '2');
+        expect.soft(stripe).toHaveAttribute('height', '200');
+      });
+    });
+  });
+
+  describe('layout and size when inherited from parent chart', () => {
+    const exampleMargin: Margin = {
+      bottom: 1,
+      left: 2,
+      right: 4,
+      top: 8,
+    };
+    it('should put x, y, width and height as coordinates to the background', () => {
+      const { container } = render(
+        <AreaChart margin={exampleMargin} width={500} height={400}>
```

Comment:
I would suggest to always use the ComposedChart, since it is the most versatile. Or, we could make this a parametrised tests and simply test for all charts?

Reply:
This was easier than I expected